internal/condition/condition_test.go

```diff
@@ -500,6 +504,120 @@ func TestCastContextToTypedParameters(t *testing.T) {
 	}
 }
 
+func TestEvaluateWithInterruptCheckFrequency(t *testing.T) {
+	makeItems := func(size int) []interface{} {
+		items := make([]interface{}, size)
+		for i := int(0); i < size; i++ {
+			items[i] = i
+		}
+		return items
+	}
+
+	var tests = []struct {
+		name           string
+		condition      *openfgav1.Condition
+		context        map[string]any
+		checkFrequency uint
+		result         condition.EvaluationResult
+		err            error
+	}{
+		{
+			name: "operation_interrupted_one_comprehension",
+			condition: &openfgav1.Condition{
+				Name:       "condition1",
+				Expression: "items.map(i, i * 2).size() > 0",
+				Parameters: map[string]*openfgav1.ConditionParamTypeRef{
+					"items": {
+						TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_LIST,
+						GenericTypes: []*openfgav1.ConditionParamTypeRef{
+							{
+								TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_INT,
+							},
+						},
+					},
+				},
+			},
+			checkFrequency: 100,
+			context: map[string]interface{}{
+				"items": makeItems(3000),
+			},
+			err: fmt.Errorf("failed to evaluate relationship condition: 'condition1' - failed to evaluate condition expression: operation interrupted"),
+		},
+		{
+			name: "operation_interrupted_two_comprehensions",
+			condition: &openfgav1.Condition{
+				Name:       "condition1",
+				Expression: "items.map(i, i * 2).map(i, i * i).size() > 0",
+				Parameters: map[string]*openfgav1.ConditionParamTypeRef{
+					"items": {
+						TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_LIST,
+						GenericTypes: []*openfgav1.ConditionParamTypeRef{
+							{
+								TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_INT,
+							},
+						},
+					},
+				},
+			},
+			checkFrequency: 100,
+			context: map[string]interface{}{
+				"items": makeItems(100),
+			},
+			err: fmt.Errorf("failed to evaluate relationship condition: 'condition1' - failed to evaluate condition expression: operation interrupted"),
+		},
+		{
+			name: "operation_not_interrupted",
+			condition: &openfgav1.Condition{
+				Name:       "condition1",
+				Expression: "items.map(i, i * 2).map(i, i * i).size() > 0",
+				Parameters: map[string]*openfgav1.ConditionParamTypeRef{
+					"items": {
+						TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_LIST,
+						GenericTypes: []*openfgav1.ConditionParamTypeRef{
+							{
+								TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_INT,
+							},
+						},
+					},
+				},
+			},
+			checkFrequency: 1000,
```

Comment:
What if we modify these tests to account for more of the boundary cases? What happens?

e.g. `checkFrequency: 100` and list comprehends over `101` items. `checkFrequency: 100` and list comprehends over `99` items. etc..

Have we tested those cases?

Reply:
updated these